crates/components-runtime/src/config.rs

```diff
@@ -0,0 +1,80 @@
+use std::{collections::HashMap, path::PathBuf};
+
+use serde::Deserialize;
+
+#[derive(Deserialize, Debug, Default, Clone)]
+pub struct ComponentsConfiguration {
+    #[serde(default = "default_data_collection_components")]
+    pub data_collection: Vec<DataCollectionComponents>,
+    #[serde(default = "default_consent_mapping_components")]
+    pub consent_mapping: Vec<ConsentMappingComponents>,
```

Comment:
On this case, since the default is an empty `Vec`, you can simply use `#[serde(default)]` here so it will use the `std::default::Default` trait which is an empty vec for this type :)

```suggestion
    #[serde(default)]
    pub data_collection: Vec<DataCollectionComponents>,
    #[serde(default)]
    pub consent_mapping: Vec<ConsentMappingComponents>,
```